### PR TITLE
Don't set order by widgets size in the layer, use param order

### LINKED ISSUE #8892: The widgets lost the order and there is no way to sort them out
Sample: https://team.carto.com/u/abel/builder/44999f04-49ca-11e6-89ee-0e233c30368f/embed?

Once the order is lost, you can't get the order back and the consequent changes are not reflected in the embed. There is no matter if you reorder and publish several times, the embed is frozen in an unordered state.

cc @saleiva

### PR DESCRIPTION
**Pull request by xavijam**
Fixes #8892.
Code by @javitonino, CR: @juanignaciosl 
cc @AbelVM

### CODE REVIEW

File: app/controllers/carto/api/widgets_controller.rb [hunk 1]
```diff
@@ -17,9 +17,15 @@ def show
       end
 
       def create
+        order_param = params[:order]
+        order = if order_param.present?
+                  order_param
+                else
+                  @map.visualization.widgets.count + 1
```
Comment (juanignaciosl) [inline app/controllers/carto/api/widgets_controller.rb]: This changes the ordering from layer-based to visualization-based. I guess that this is what we actually want, but I wanted to be sure, as ordering has been used for identification of entities within collections in the past.

File: spec/requests/carto/api/widgets_controller_spec.rb [hunk 2]
```diff
@@ -206,23 +223,6 @@ def widget_payload(
         response.status.should == 403
       end
     end
-
-    it 'assigns consecutive orders for widgets for the same layer' do
```
Comment (juanignaciosl) [inline spec/requests/carto/api/widgets_controller_spec.rb]: Do you need to actually delete this?

Comment (javitonino) [inline spec/requests/carto/api/widgets_controller_spec.rb]: "Aggressive" optimization :stuck_out_tongue:. I'll try and get it restored.

**Comment by xavijam** [comment]
I feel this ticket all yours @javitonino, re-assigning the ticket. 💋

File: app/controllers/carto/api/widgets_controller.rb (hunk 1 above)
Comment (javitonino) [inline app/controllers/carto/api/widgets_controller.rb]: In fact, it should not normally go this way as the frontend always send the order. But yes, I think the idea is that order should be per-visualization, as all widgets of the vis are shown together. Probably @xavijam can give additional details if needed.

Rethinking this, I'm tempted to remove the `+ 1` to generate the orders the same way as the frontend (0-based).

Comment (juanignaciosl) [inline app/controllers/carto/api/widgets_controller.rb]: +1 to not -1 (same order than frontend, 0-based).

**Comment by javitonino** [comment]
Tests are back to :green_apple: Round 2!

**Comment by juanignaciosl** [comment]
✨

**Pull request merged**